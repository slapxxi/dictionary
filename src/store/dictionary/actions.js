// @flow
import type { Action, ID } from '../types';

function view(id: ID): Action {
  return { type: 'VIEW', payload: id };
}

function learn(id: number): Action {
  return { type: 'LEARN', payload: id };
}

function changeIndex(index: ID): Action {
  return { type: 'CHANGE_INDEX', payload: index };
}

export { learn, changeIndex, view };
