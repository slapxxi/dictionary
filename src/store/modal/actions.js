function openModal() {
  return { type: 'OPEN_MODAL' };
}

function closeModal() {
  return { type: 'CLOSE_MODAL' };
}

export { openModal, closeModal };
