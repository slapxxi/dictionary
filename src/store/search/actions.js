// @flow
import type { Action, Query } from '../types';

function search(query: Query): Action {
  return { type: 'SEARCH', payload: query };
}

export { search };
