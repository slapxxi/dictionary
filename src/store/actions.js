// @flow
import { openModal, closeModal } from './modal/actions';
import { learn, changeIndex, view } from './dictionary/actions';
import { search } from './search/actions';

export { openModal, closeModal, learn, changeIndex, search, view };
