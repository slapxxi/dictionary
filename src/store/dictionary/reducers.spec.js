import reducer from './reducers';
import { view, changeIndex, learn } from './actions';

const mockState = {
  index: 0,
  entries: [{ id: 0, viewCount: 0 }, { id: 1, viewCount: 0 }],
};

describe('when changing index', () => {
  it('changes index', () => {
    const store = reducer(mockState, changeIndex(1));
    expect(store).toMatchSnapshot();
  });

  it('does not change index if beyond limits', () => {
    const store = reducer(mockState, changeIndex(2));
    expect(store).toMatchSnapshot();
  });
});

describe('when viewed', () => {
  it('increases view count of item with matching id', () => {
    const store = reducer(mockState, view(0));
    expect(store).toMatchSnapshot();
  });

  it('ignores view count non matching items', () => {
    const store = reducer(mockState, view(2));
    expect(store).toMatchSnapshot();
  });
});

describe('when learning', () => {
  it('learns matching word', () => {
    const store = reducer(mockState, learn(0));
    expect(store).toMatchSnapshot();
  });

  it('ignores non matching words', () => {
    const store = reducer(mockState, learn(2));
    expect(store).toMatchSnapshot();
  });
});
