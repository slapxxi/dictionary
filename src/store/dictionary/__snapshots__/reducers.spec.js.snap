// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when changing index changes index 1`] = `
Object {
  "entries": Array [
    Object {
      "id": 0,
      "viewCount": 0,
    },
    Object {
      "id": 1,
      "viewCount": 0,
    },
  ],
  "index": 1,
}
`;

exports[`when changing index does not change index if beyond limits 1`] = `
Object {
  "entries": Array [
    Object {
      "id": 0,
      "viewCount": 0,
    },
    Object {
      "id": 1,
      "viewCount": 0,
    },
  ],
  "index": 0,
}
`;

exports[`when learning ignores non matching words 1`] = `
Object {
  "entries": Array [
    Object {
      "id": 0,
      "viewCount": 0,
    },
    Object {
      "id": 1,
      "viewCount": 0,
    },
  ],
  "index": 0,
}
`;

exports[`when learning learns matching word 1`] = `
Object {
  "entries": Array [
    Object {
      "id": 0,
      "learnt": true,
      "viewCount": 0,
    },
    Object {
      "id": 1,
      "viewCount": 0,
    },
  ],
  "index": 0,
}
`;

exports[`when viewed ignores view count non matching items 1`] = `
Object {
  "entries": Array [
    Object {
      "id": 0,
      "viewCount": 0,
    },
    Object {
      "id": 1,
      "viewCount": 0,
    },
  ],
  "index": 0,
}
`;

exports[`when viewed increases view count of item with matching id 1`] = `
Object {
  "entries": Array [
    Object {
      "id": 0,
      "viewCount": 1,
    },
    Object {
      "id": 1,
      "viewCount": 0,
    },
  ],
  "index": 0,
}
`;
